File: TEST/src/AMSuite2.Web/src/app/agenda-manager/meeting/meeting.model.ts
import { BaseEntity, inArray, DateHelper } from '@caiu/core';
import { RouterState } from '@caiu/router';

import { AgendaItem } from '../agenda-items/agenda-items.model';
import { Agenda } from '../agendas/agendas.model';
import { Account } from '../../account/account.model';
import { EmailItem } from '../../core/email/email.model';
import { GroupMember, Group } from '../../core/groups/groups.model';
import { SecurityStatus } from '../../shared/models';

export class Meeting extends BaseEntity {
    accountId = 0;
    attendanceTakerId = 0;
    comments = '';
    conferenceId = 0;
    date: Date = new Date();
    endTime: Date = new Date();
    groupId = 0;
    location = '';
    markedForDelete: Date = new Date();
    minuteTakerId = 0;
    name = '';
    startTime: Date = new Date();
    securityStatusId = 0;
    typeId = 0;
    voteTakerId = 0;

    account: Account = new Account();
    agendas: Agenda[] = [];
    durationHour = 0;
    durationMinutes = 0;
    email: EmailItem[] = [];
    generalInfo = '';
    groupMembers: GroupMember[] = [];
    groupName = '';
    groups: Group[] = [];
    securityStatusName = '';
    startHour = 0;
    startMinutes = 0;
    startMeridian: 'AM' | 'PM' = 'AM';

    get accountName(): string {
        return this.account.name;
    }

    set accountName(value: string) {
        if (this.account) {
            this.account.name = value;
        }
    }
}

File: DEV/src/AMSuite2.API/Features/AgendaManager/Meetings/MeetingModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using AMSuite2.API.Features.AgendaManager.Agendas;
using AMSuite2.API.Features.Core.Accounts;
using AMSuite2.API.Features.Core.Email;
using AMSuite2.API.Features.Core.GroupMembers;
using AMSuite2.API.Infrastructure.Models;

namespace AMSuite2.API.Features.AgendaManager.Meetings
{
    public class MeetingModel : BaseEntity, IEntity
    {
        public int Id { get; set; }
        public int AccountId { get; set; }
        public int? AttendanceTakerId { get; set; }
        public string Comments { get; set; }
        public int? ConferenceId { get; set; }
        public DateTime Date { get; set; }
        public TimeSpan? EndTime { get; set; }
        public int? GroupId { get; set; }
        public string Location { get; set; }
        public DateTime? MarkedForDelete { get; set; }
        public int? MinuteTakerId { get; set; }
        public string Name { get; set; }
        public int? OutlineId { get; set; }
        public TimeSpan? StartTime { get; set; }
        public int SecurityStatusId { get; set; }
        public int? TypeId { get; set; }
        public int? VoteTakerId { get; set; }

        public AccountModel Account { get; set; }
        public string AccountName { get; set; }
        public IEnumerable<AgendaModel> Agendas { get; set; }
        public IEnumerable<EmailModel> Email { get; set; }
        public IEnumerable<GroupMemberModel> GroupMembers { get; set; }
        public string GroupName { get; set; }
        public string SecurityStatusName { get; set; }
    }

    public class MeetingFiltersModel
    {
        public DateTime StartDate { get; set; } = DateTime.Today.AddDays(-100);
        public DateTime EndDate { get; set; } = DateTime.Today.AddDays(100);
        public int TakeNumber { get; set; } = 200;
        public int GroupId { get; set; } = 0;
        public int UserId { get; set; } = 38;
        public int AccountId { get; set; } = 4;
    }

    public class MeetingsQueryModel : QueryModel<MeetingModel>
    {
        public DateTime StartDate { get; set; } = DateTime.Today.AddDays(-100);
        public DateTime EndDate { get; set; } = DateTime.Today.AddDays(100);
        public int GroupId { get; set; } = 0;
    }
}
